backend/de.metas.contracts/src/main/java/de/metas/contracts/commission/salesrep/process/CreateMissingCommissionEntriesForSalesRep.java

```diff
@@ -0,0 +1,432 @@
+	protected String doIt() throws Exception
+	{
+		try
+		{
+			final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+			final ZonedDateTime from = LocalDate.parse(FROM_DATE, formatter).atStartOfDay(ZoneId.of("UTC-8"));
+			final ZonedDateTime to = LocalDate.parse(TO_DATE, formatter).atStartOfDay(ZoneId.of("UTC-8"));
+
+			final InstantInterval targetInterval = InstantInterval.of(from.toInstant(), to.toInstant());
+
+			recalculateCommission(targetInterval);
+		}
+		catch (final Exception e)
+		{
+			Loggables.withLogger(logger, Level.ERROR).addLog("*** ERROR: {}", e.getMessage(), e);
+		}
+
+		return MSG_OK;
+	}
+
+	private void recalculateCommission(@NonNull final InstantInterval targetInterval)
+	{
+		final SalesRepRelatedBPTableInfo salesRepRelatedBPTableInfo = getSalesRepRelatedBPTableInfo();
+
+		final ImmutableSet<BPartnerId> preliminaryQualifiedBPIds = retrieveAllBPsWithASalesRep(targetInterval, salesRepRelatedBPTableInfo);
+
+		Loggables.withLogger(logger, Level.DEBUG)
+				.addLog("*** DEBUG: number of preliminary qualified bPartnerIds for recalculation={}", preliminaryQualifiedBPIds.size());
+
+		long count = 0;
+		long countError = 0;
+		for (final BPartnerId bPartnerId : preliminaryQualifiedBPIds)
+		{
+			try
+			{
+				count++;
+				processBPId(bPartnerId, salesRepRelatedBPTableInfo, targetInterval);
+			}
+			catch (final Exception e)
+			{
+				Loggables.withLogger(logger, Level.ERROR).addLog("*** ERROR while trying to calculate commission for bpartnerId: {}, message: {}"
+						, bPartnerId, e.getMessage(), e);
+				countError++;
+			}
+		}
+		Loggables.withLogger(logger, Level.DEBUG)
+				.addLog("*** DEBUG: iterated {} preliminary qualified bPartnerIds; countError={}", count, countError);
+	}
+
+	private void recalculateForBPartnerId(
+			@NonNull final BPartnerId bPartnerId,
+			@NonNull final ImmutableList<ChangeLogEntry> salesRepChangeLogEntries,
+			@NonNull final ImmutableList<ChangeLogEntry> bPartnerSalesRepChangeLogEntries,
+			@NonNull final InstantInterval targetTimeframe)
+	{
+		Loggables.withLogger(logger, Level.DEBUG)
+				.addLog("*** DEBUG: start recalculating for bpartnerId: {}! ", bPartnerId);
+
+		final List<RecalculateCommissionCriteria> recalculateCommissionCriteriaList =
+				extractCommissionCriteriaList(bPartnerId, salesRepChangeLogEntries, bPartnerSalesRepChangeLogEntries, targetTimeframe);
+
+		Loggables.withLogger(logger, Level.DEBUG)
+				.addLog("*** DEBUG: recalculateCommissionCriteriaList {}! ", recalculateCommissionCriteriaList);
+
+		final I_C_BPartner bPartner = bPartnerDAO.getById(bPartnerId);
+		final BPartnerId initialSalesRepBPartnerId = BPartnerId.ofRepoIdOrNull(bPartner.getC_BPartner_SalesRep_ID());
+		try
+		{
+			recalculateCommissionCriteriaList.forEach(criteria -> {
+				try
+				{
+					trxManager.runInNewTrx(() -> recalculateCommissionForCriteria(criteria));
+				}
+				catch (final RuntimeException e)
+				{
+					Loggables.withLogger(logger, Level.ERROR)
```

Comment:
Use IErrorManager#createIssue(java.lang.Throwable) to create an AD_Issue and log its AD_Issue_ID as well